### ISSUE TITLE
[Bug] API sends all hidden states when asked for only mean hidden state

### ISSUE THREAD

**Issue by kumarshreshtha** [evolutionaryscale/esm#292]
I haven't tested it for other models but for ESM-C-6B when the logits config is set to `logits_config = esm_api.LogitsConfig(sequence=True, return_mean_hidden_states=True)` the api also returns the full hidden states of all layers. This makes the api calls really slow.

Since the expected behaviour for the 6B model is to never send all hidden states this seems to be a bug in how the LogitsConfig is handled.

**Comment by imathur1** [comment]
Are you still observing this issue? `return_mean_hidden_states` returns hidden states mean pooled along the sequence length (L) dimension, not the hidden state dimension (`n_layers`). So `return_mean_hidden_states` would have shape `[B, n_layers+1, D]`
